packages/blitz-next/src/index-server.ts

```diff
@@ -50,13 +58,47 @@ export const setupBlitzServer = ({plugins}: SetupBlitzOptions) => {
         (y, f) => (f ? f(y) : y),
         (res as MiddlewareResponse).blitzCtx,
       )
-      return handler({req, res, ctx, ...rest})
+      let queryClient: null | QueryClient = null
+
+      ctx.prefetchBlitzQuery = async (fn, input, defaultOptions = {}) => {
+        queryClient = new QueryClient({defaultOptions})
+
+        const queryKey = getQueryKey(fn, input)
+        await queryClient.prefetchQuery(queryKey, () => fn(input, ctx))
+      }
+
+      const result = await handler({req, res, ctx, ...rest})
+      if ("props" in result) {
+        const props = {
+          ...result.props,
+          ...(queryClient ? {dehydratedProps: dehydrate(queryClient)} : {}),
+        }
+        return {...result, props}
+      }
```

Comment:
why do we only add dehydratedProps if `"props" in result`?

Reply:
Ah good point, will change that